### PR TITLE
Ansible - WordPress

### PR DESCRIPTION
**Pull request by rojinebrahimi**
I'd be grateful if you could mention the improvements I can apply on my code. Thanks.

### CODE REVIEW

File: vars.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
```
Comment (korney4eg) [inline vars.yaml]: +1 for using variables file

File: wordpress.yaml [hunk 2]
```diff
@@ -1,7 +1,152 @@
...
+  
+    - name: Set root password in MYSQL
+      mysql_user:
+        name: root
+        password: "{{ mysql_password }}"
+        login_user: root
+        login_password: "{{ mysql_password }}"
+        login_unix_socket: "/var/run/mysqld/mysqld.sock"
+
+    - name: Remove all MYSQL anonymous users
```
Comment (korney4eg) [inline wordpress.yaml:72]: nice!

File: wordpress.yaml [hunk 3]
```diff
@@ -1,7 +1,152 @@
...
+    - name: Create WordPress user
+      mysql_user:
+        name: "wordpressuser"
+        password: "{{ wp_database_password }}"
+        priv: "*.*:ALL,GRANT"
+        state: present
+        login_user: root
+        login_password: "{{ mysql_password }}"
+    
+    - name: Check WordPress directory existance
```
Comment (korney4eg) [inline wordpress.yaml]: what is the reason for this check here?

File: wordpress.yaml [hunk 4]
```diff
@@ -1,7 +1,152 @@
...
+        path: "/var/www/{{ wp_conf_dir }}"
+      register: wp_dir
+
+    - name: Create directory
+      file:
+        path: "/var/www/{{ wp_conf_dir }}"
+        state: directory
+      when: not wp_dir.stat.exists
+
+    - name: Check WordPress file existance
```
Comment (korney4eg) [inline wordpress.yaml]: what is the reason for this check here?

File: wordpress.yaml [hunk 5]
```diff
@@ -1,7 +1,152 @@
...
+        path: "/tmp/{{ wp_file }}"
+      register: wp
+
+    - name: Download WordPress zip
+      get_url:
+        url: "https://wordpress.org/{{ wp_file }}"
+        dest: /tmp/
+      when: not wp.stat.exists
+
+    - name: Extract WordPress
```
Comment (korney4eg) [inline wordpress.yaml:113]: here you try to extract archive every run

File: wordpress.yaml [hunk 6]
```diff
@@ -1,7 +1,152 @@
...
+        dest: "/var/www/{{ wp_conf_dir }}/wp-config.php"
+        
+    - name: Install WP-CLI
+      get_url:
+        url: "{{ wp_cli_url }}"
+        dest: /tmp/
+
+    - name: Set WP-CLI Permission
+      shell: |
+        chmod +x /tmp/{{ wp_cli_file }}
```
Comment (korney4eg) [inline wordpress.yaml]: it's work, but there is a more elegant way to do this with ansible

File: wordpress.yaml [hunk 7]
```diff
@@ -1,7 +1,152 @@
...
+        
+    - name: Install WP-CLI
+      get_url:
+        url: "{{ wp_cli_url }}"
+        dest: /tmp/
+
+    - name: Set WP-CLI Permission
+      shell: |
+        chmod +x /tmp/{{ wp_cli_file }} 
+        mv /tmp/{{ wp_cli_file }} /usr/local/bin/wp
```
Comment (korney4eg) [inline wordpress.yaml]: why do you need to move file from one place to another, not just download this into desired folder?

File: wordpress.yaml [hunk 8]
```diff
@@ -1,7 +1,152 @@
...
+        - php-xmlrpc
+        - php-zip
+          
+    - name: Configure firewall
+      ufw:
+        rule: allow
+        name: "Nginx Full"
+        state: enabled
+    
+    - name:
```
Comment (korney4eg) [inline wordpress.yaml]: +1 for using template. But what is the reason to use template here? :)

**Review by korney4eg — COMMENTED** [review]
good first try

File: wordpress.yaml (hunk 4 above)
Comment (rojinebrahimi) [reply wordpress.yaml]: I thought if the file is already downloaded, it's better to use the available one and avoid extra traffic.

File: wordpress.yaml (hunk 3 above)
Comment (rojinebrahimi) [reply wordpress.yaml]: maybe if somehow provisioning gets corrupted or while doing it again, it won't create the directory again. however, I'm not sure if I can also check it with its state or not

File: wordpress.yaml (hunk 5 above)
Comment (rojinebrahimi) [reply wordpress.yaml:113]: Oh, I'll try to refine it.

File: wordpress.yaml (hunk 6 above)
Comment (rojinebrahimi) [reply wordpress.yaml]: I will check out other solutions as well

File: wordpress.yaml (hunk 7 above)
Comment (rojinebrahimi) [reply wordpress.yaml]: Yes, I'll think of the solution

File: wordpress.yaml (hunk 8 above)
Comment (rojinebrahimi) [reply wordpress.yaml]: I thought I could later use some values of the ansible in the conf file. But I will read more about it.

Comment (korney4eg) [reply wordpress.yaml]: that make sence later with variables

File: wordpress.yaml (hunk 4 above)
Comment (korney4eg) [reply wordpress.yaml]: this is a main reason of Configuration Management and regular scripts. CM for remote_url will check if it hasn't already downloaded

File: wordpress.yaml (hunk 5 above)
Comment (korney4eg) [reply wordpress.yaml:113]: Hey, what would be if you use `creates` here?

File: wordpress.yaml [hunk 9]
```diff
@@ -1,7 +1,158 @@
...
+        dest: "/var/www/{{ wp_conf_dir }}/wp-config.php"
+        
+    - name: Install WP-CLI
+      get_url:
+        url: "{{ wp_cli_url }}"
+        dest: "/usr/local/bin/wp"
+        mode: 0755
+
+    - name: Change directory to /var/www/wordpress and install WP
+      command: wp core install --url="192.168.50.2" --title="WordPress Setup" --admin_user=admin --admin_email=admin@test.com --admin_password="!2three456." --path=/var/www/wordpress --allow-root
```
Comment (korney4eg) [inline wordpress.yaml]: not a mistake, but you have variable with `192.168.50.2` why not using it?

BTW, also as you did with MySQL credentials, maybe it make sence to use Wordpress configurations from `variables.yaml` file?

File: wp-config.php [hunk 10]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+/** The name of the database for WordPress */
+define( 'DB_NAME', 'wordpress' );
+
+/** MySQL database username */
+define( 'DB_USER', 'wordpressuser' );
+
+/** MySQL database password */
+define( 'DB_PASSWORD', 'password' );
```
Comment (korney4eg) [inline wp-config.php]: I think this config file could be also converted into template and you woun't need to dublicate db_user, bassword, db_name here and during database creation

**Review by korney4eg — COMMENTED** [review]
I like your ideas and will take idea with using `variables.yaml` for everybody.
Overall looks good. Some fixes and it become perfect

File: wp-config.php (hunk 10 above)
Comment (rojinebrahimi) [reply wp-config.php]: yes, I'll change the wp-conf.php

File: wordpress.yaml (hunk 9 above)
Comment (rojinebrahimi) [reply wordpress.yaml]: I can add them. will refine it

File: wordpress.yaml [hunk 11]
```diff
@@ -1,7 +1,153 @@
...
+    - name: Create new database
+      mysql_db:
+        name: "wordpress"
+        state: present
+        login_user: root
+        login_password: "{{ mysql_password }}"
+
+    - name: Create WordPress user
+      mysql_user:
+        name: "wordpressuser"
```
Comment (korney4eg) [inline wordpress.yaml]: why is this hardcoded?

Comment (rojinebrahimi) [reply wordpress.yaml]: now using credentials in variables.yaml

**Pull request closed without merging**